pkg/controller/production_test.go

```diff
@@ -93,11 +97,15 @@ func TestDelegateProductionDepedencies(t *testing.T) {
 	assert.Equal(t, c.Dependencies(), storage.Dependencies(c.jaeger))
 }
 
-func assertDeploymentsAndServicesForProduction(t *testing.T, name string, objs []sdk.Object, hasDaemonSet bool) {
+func assertDeploymentsAndServicesForProduction(t *testing.T, name string, objs []sdk.Object, hasDaemonSet bool, hasOAuthProxy bool) {
+	expectedNumObjs := 5
+
 	if hasDaemonSet {
-		assert.Len(t, objs, 6)
-	} else {
-		assert.Len(t, objs, 5)
+		expectedNumObjs++
+	}
+
+	if hasOAuthProxy {
+		expectedNumObjs++
```

Comment:
value of expectedNumObjs is not being asserted

Reply:
Nice catch!